pkg/resourcemanager/webhook/podzoneaffinity/handler.go

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package podzoneaffinity
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	gardencorev1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	admissionv1 "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+var podGVK = metav1.GroupVersionKind{Group: "", Kind: "Pod", Version: "v1"}
+
+type handler struct {
+	client  client.Client
+	decoder *admission.Decoder
+}
+
+// NewHandler returns a new handler.
+func NewHandler() admission.Handler {
+	return &handler{}
+}
+
+func (h *handler) InjectClient(cl client.Client) error {
+	h.client = cl
+	return nil
+}
+
+func (h *handler) InjectDecoder(d *admission.Decoder) error {
+	h.decoder = d
+	return nil
+}
+
+func (h *handler) Handle(ctx context.Context, req admission.Request) admission.Response {
```

Comment:
Please add an integration test to https://github.com/gardener/gardener/tree/master/test/integration/resourcemanager